src/Toolkit/Str.php

```diff
@@ -528,6 +528,21 @@ public static function from(string $string, string $needle, bool $caseInsensitiv
         return static::substr($string, $position);
     }
 
+    /**
+     * Add's _1 to a string or increment the ending number to allow _2, _3, etc
+     *
+     * @param string $string The string to increment
+     * @param int $first Start with
+     * @param string $separator Separator
+     * @return string
+     */
+    public static function increment(string $string, int $first = 1, string $separator = '_'): string
+    {
+        preg_match('/(.+)' . $separator . '([0-9]+)$/', $string, $match);
```

Comment:
To ensure that any separator will work, even if it's a special character in RegEx:

```suggestion
        preg_match('/(.+)' . preg_quote($separator, '/') . '([0-9]+)$/', $string, $match);
```